src/main/com/yetanalytics/pan/axioms.cljc

```
@@ -57,42 +56,43 @@
 ;; cases. The main exception is with the pipe character '|', which, if placed
 ;; before the '$' character separates two JSONPath strings (unless escaped).
 ;; See: regexr.com/4fk7v
-;;
+
 ;; JSONPath regexes
 (def JSONPathRegEx #"\$((((\.\.?)([^\[\]\.,\s]+|(?=\[)))|(\[\s*(('([^,']|(\\\,)|(\\\'))+'(,\s*('([^,']|(\\\,)|(\\\'))+'))*\s*)|\*)\s*\]))(\[((\d*(,\s*(\d*))*)|\*)\])?)*")
 (def JSONPathSplitRegEx #"\s*\|\s*(?!([^\[]*\]))")
 
+; Need to filter out nils caused by JavaScript regexes
 (s/def ::json-path
   (s/and ::string
          (fn [paths]
            (every? some?
                    (map (partial re-matches JSONPathRegEx)
-                        (#(string/split % JSONPathSplitRegEx) paths))))))
+                        (filterv some? ; Filter out nils from JS regexes
+                                 (string/split paths JSONPathSplitRegEx)))))))
 
 ;; JSON Schema
 ;; Example: "{\"type\":\"array\", \"uniqueItems\":true}"
-;; 
 
-;; FIXME: add additional json schema draft versions as needed from https://json-schema.org/specification-links.html
-;; - add to resources/json
-
-(defn schema-validate [schema json]
-  (try (do (js/validate schema json) true)
-       (catch Exception e false)))
+(defn schema-validate [json-schema json]
+  #?(:clj
+     (try (let [_ (jschema/validate json-schema json)] true)
+          (catch Exception _ false))
```

does `jschema/validate` throw when invalid? what's returned when valid?

---

From their [Github](https://github.com/luposlip/json-schema):
> All found errors will cause the function to throw an instance of `clojure.lang.ExceptionInfo`

---

It also just returns the validated data if there are no errors.

---

kk please adjust this to check the return of `jschema/validate` for validity instead of just returning true bc rn everything is valid unless it causes `jschema/validate` to throw

---

Isn't that what we want though? If `jschema/validate` throws an exception, then the schema is invalid, right?

---

right my issue isn't with `catch Exception _ false` its with `([_ (jschema/validate json-schema json)] true)`

The return of `jschema/validate` is never used and the update can be super simple:

`([r (jschema/validate json-schema json)] (some? r))`

I understand that functionally, there is no difference but the update makes the code more readable/understandable without knowing anything about the impl of `jschema/validate`

---

Ok. I refactored the clj code to look more like the cljs code.